Gravity hammer balance and ODST call in fix

OuterMostMonk opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->
Attempts to make the powered gravity hammer useable but not too strong. Also fixes ODSTs having more than just personalised call ins.
Hopefully could lead to Chieftain being enabled in the future whether as an alternative to shipmaster or as an extra.

<!-- If you need a change log update the below. Other wise you should remove it-->
:cl: OuterMonk
tweak: Powered gravity hammer no longer spammable, hunters immune to its knockback.
tweak: Non officer ODSTs get personalised call downs only
/:cl:

OuterMostMonk committed:
Balances gravity hammer

Makes the powered hammer less broken by preventing killing enemies by clicking as fast as possible. Prevents instant door breaking and pushing back of hunters with it.

OuterMostMonk committed:
Chieftain armor fix

Sets the max armour properly so chieftains don't seem to have 200% armour displayed when examined.

OuterMostMonk committed:
Chieftain hammer spawn

Gives chieftain job its hammer back (Since it hopefully should not be busted now)

OuterMostMonk committed:
Reduces ODST supply call down rank

Normal ODSTs were able to call down medical and other supplies they shouldn't. They are only intended to have personalised drops.
Officers retain their current support options.

OuterMostMonk committed:
Forgot hunter check

Adds a check I forgot to put in. To stop direct hits moving the hunter.

X0-11 commented on code/modules/halo/weapons/covenant/brute.dm:
```diff
@@ -265,23 +266,26 @@
 
 	var/atom/throw_target = get_edge_target_turf(A, get_dir(user, A))
-	if(istype(A, /atom/movable))
+	if(istype(A, /atom/movable) && !istype(A, /obj/machinery/door) && !istype(A, /mob/living/simple_animal/mgalekgolo)) //No instant door breaking or bullying poor hunters
```
This could probably just be narrowed down to only shift /mob/living and /obj/item/

X0-11 commented on code/modules/halo/weapons/covenant/brute.dm:
```diff
@@ -265,23 +266,26 @@
-	for(var/atom/movable/M in range(A,1))
-		if(M == user)
-			continue
+	if((world.time-timer)>29) // To avoid spamming non direct hits
```
Use a define instead of a magic number like this.

X0-11 commented on code/modules/halo/weapons/covenant/brute.dm:
```diff
@@ -265,23 +266,26 @@
-		if(M == user)
-			continue
+	if((world.time-timer)>29) // To avoid spamming non direct hits
+		timer = world.time + 30
```
The define can be used here too.

OuterMostMonk committed:
Cleaner code

Attempts to make the code less messy

OuterMostMonk committed:
I forgot the define lol

No wonder it was failing

OuterMostMonk commented:
Changes made, should be better now (I hope). I kept hunters immune with the narrowing down of targets but wonder if complete immunity is too much. Maybe I should allow them to be moved by the ground attack which has a cooldown but not direct attacks?

X0-11 commented:
I think keeping hunters immune is fine as-is, although you may want to check through the code and see if there's some internal size tracking for simplemobs that you can check against rather than specifically targeting hunters, as flood juggernauts should also probably be immune.

OuterMostMonk committed:
Proc for checking mob size

For use with checks in the hammer (couldn't find any existing ways to check if the mob is large)

OuterMostMonk committed:
Uses a proc to forbid large mobs from being thrown

Should prevent both Juggernauts and Hunters from being thrown by the hammer (And any other mob with a large size)

OuterMostMonk committed:
Merge branch 'HaloSpaceStation:alpha' into hammer

OuterMostMonk commented:
Large size mobs are now immune to gravity hammer knockback. Not sure what other mobs count as large but I imagine for halo mobs it is just juggernaut and hunter.
